test/pkg/integration/workspace.go

```diff
@@ -375,8 +432,31 @@ func (it *Test) WaitForWorkspaceStop(instanceID string) (lastStatus *wsmanapi.Wo
 	select {
 	case <-it.ctx.Done():
 		it.t.Fatalf("cannot wait for workspace: %q", it.ctx.Err())
+		return
 	case <-done:
 	}
+
+	// wait for the Theia service to be properly deleted
+	ctx, cancel := context.WithTimeout(it.ctx, 30*time.Second)
+	defer cancel()
+	var (
+		start       = time.Now()
+		serviceGone bool
+		k8s, ns     = it.API().Kubernetes()
+	)
+	for time.Since(start) < 1*time.Minute {
+		_, err := k8s.CoreV1().Services(ns).Get(ctx, fmt.Sprintf("ws-%s-theia", workspaceID), v1.GetOptions{})
+		if err != nil {
```

Comment:
this should check for `errors.IsNotFound(err)`
https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors?utm_source=godoc#IsNotFound